src/pip/_internal/utils/misc.py

```diff
@@ -852,17 +853,34 @@ def enum(*sequential, **named):
     return type('Enum', (), enums)
 
 
-def remove_auth_from_url(url):
-    # Return a copy of url with 'username:password@' removed.
-    # username/pass params are passed to subversion through flags
-    # and are not recognized in the url.
+def redact_password_from_url(url):
+    """Return a copy of the url by redacting the password in case it was
+    present. The password will be replaced with '****'.
+    """
+    purl = urllib_parse.urlsplit(url)
+
+    redacted_netloc = purl.netloc
+    if purl.password:
+        auth, netloc = redacted_netloc.split('@')
```

Comment:
FYI, this will cause an exception if the user included a password with an unencoded "@", like
"https://user:pass@word@domain.tld/svn/project/trunk"